### PR TITLE
refactor(functions): support null values in percentile

### LINKED ISSUE #660: Update percentile to understand Null

### PR DESCRIPTION
**Pull request by jsternberg**
- [x] docs/SPEC.md updated
- [x] Test cases written

Fixes #660.

### CODE REVIEW

File: functions/transformations/percentile.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 	"math"
 	"sort"
 
+	"github.com/influxdata/flux/values"
```
Comment (wolffcm) [inline functions/transformations/percentile.go]: This import is out of order.

File: functions/transformations/percentile_test.go [hunk 2]
```diff
@@ -374,6 +374,33 @@ func TestPercentileSelector_Process(t *testing.T) {
...
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_value", Type: flux.TFloat},
+					{Label: "t1", Type: flux.TString},
+					{Label: "t2", Type: flux.TString},
+				},
+				Data: [][]interface{}{
+					{nil, nil, "a", nil},
+				},
+			}},
+		},
```
Comment (wolffcm) [inline functions/transformations/percentile_test.go]: Would be good to have a test with null values in the `_value` column in a non-empty input table.

File: functions/transformations/percentile.go (hunk 1 above)
Comment (jsternberg) [reply functions/transformations/percentile.go]: Fixed.

File: functions/transformations/percentile_test.go (hunk 2 above)
Comment (jsternberg) [reply functions/transformations/percentile_test.go]: Done.

**Pull request merged**